tests failing

Using Go 1.15.5 on OSX:
```
❯ go test
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x116535d]

goroutine 907 [running]:
github.com/multiformats/go-multistream.(*rwcStrict).Write(0xc0001f5760, 0xc0000e0000, 0x14, 0x1000, 0xc0003902f1, 0x1, 0x1)
	/Users/marten/src/go/src/github.com/multiformats/go-multistream/multistream_test.go:41 +0xdd
bufio.(*Writer).Flush(0xc0000d4680, 0xc0000d4680, 0xc000388fa0)
	/usr/local/Cellar/go/1.15.5/libexec/src/bufio/bufio.go:607 +0x7b
github.com/multiformats/go-multistream.delimWriteBuffered(0x1708fd0, 0xc0001f5760, 0xc000388fa0, 0x12, 0x12, 0x0, 0x0)
	/Users/marten/src/go/src/github.com/multiformats/go-multistream/multistream.go:81 +0xf4
github.com/multiformats/go-multistream.(*MultistreamMuxer).NegotiateLazy.func1()
	/Users/marten/src/go/src/github.com/multiformats/go-multistream/multistream.go:227 +0x13c
sync.(*Once).doSlow(0xc0000fd620, 0xc00038c840)
	/usr/local/Cellar/go/1.15.5/libexec/src/sync/once.go:66 +0xec
sync.(*Once).Do(0xc0000fd620, 0xc00038c840)
	/usr/local/Cellar/go/1.15.5/libexec/src/sync/once.go:57 +0x45
created by github.com/multiformats/go-multistream.(*MultistreamMuxer).NegotiateLazy
	/Users/marten/src/go/src/github.com/multiformats/go-multistream/multistream.go:222 +0x1ec
exit status 2
FAIL	github.com/multiformats/go-multistream	0.563s
```

--- a/multistream_test.go
+++ b/multistream_test.go
@@ -150,7 +150,7 @@ func TestProtocolNegotiationLazy(t *testing.T) {
 }
 
 func TestNegLazyStressRead(t *testing.T) {
-	count := 1000
+	const count = 100
 
 	mux := NewMultistreamMuxer()
 	mux.AddHandler("/a", nil)
@@ -203,7 +203,7 @@ func TestNegLazyStressRead(t *testing.T) {
 }
 
 func TestNegLazyStressWrite(t *testing.T) {
-	count := 1000
+	const count = 100
 
 	mux := NewMultistreamMuxer()
 	mux.AddHandler("/a", nil)

--- a/multistream_test.go
+++ b/multistream_test.go
@@ -50,21 +50,29 @@ func (s *rwcStrict) Close() error {
 func newPipe(t *testing.T) (io.ReadWriteCloser, io.ReadWriteCloser) {
 	ln, err := net.Listen("tcp", "localhost:0")
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
 	cchan := make(chan net.Conn)
+	errChan := make(chan error, 1)
 	go func() {
 		c, err := ln.Accept()
 		if err != nil {
-			t.Error(err)
+			errChan <- err
+			return
 		}
 		cchan <- c
 	}()
 	c, err := net.Dial("tcp", ln.Addr().String())
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
-	return newRwcStrict(t, <-cchan), newRwcStrict(t, c)
+	select {
+	case err := <-errChan:
+		t.Fatal(err)
+		return nil, nil
+	case rwc := <-cchan:
+		return newRwcStrict(t, rwc), newRwcStrict(t, c)
+	}
 }
 
 func TestProtocolNegotiation(t *testing.T) {

